### ISSUE TITLE
segfault with hdf4 chunking

### ISSUE THREAD

**Issue by rabernat** [Unidata/netcdf4-python#440]
(This issue was discovered through xray/xray#484)

I am trying to read data from the NASA MERRA reanalysis. An example file is: ftp://goldsmr3.sci.gsfc.nasa.gov/data/s4pa/MERRA/MAI3CPASM.5.2.0/2014/01/MERRA300.prod.assim.inst3_3d_asm_Cp.20140101.hdf
The file format is hdf4 (NOT hdf5). ([full file specification](http://gmao.gsfc.nasa.gov/pubs/docs/Lucchesi528.pdf))

```python
from netCDF4 import Dataset
nc = Dataset('MERRA300.prod.assim.inst3_3d_asm_Cp.20140101.hdf')
v = nc.variables['SLP']
v.chunking()
```
This gives me a segfault with the following stack trace:
```
/lib64/libnetcdf.so.7(NC4_inq_var_all+0x1e8)[0x7f370c4baea8]
/lib64/libnetcdf.so.7(nc_inq_var_chunking+0xb6)[0x7f370c465ae6]
/usr/local/anaconda/lib/python2.7/site-packages/netCDF4/_netCDF4.so(+0x1d930)[0x7f3730126930]
...
```
Any idea what might be wrong here?

Here is my nc-config
```
$ nc-config --all

This netCDF 4.3.2 has been built with the following features: 

  --cc        -> gcc
  --cflags    ->  -I/usr/include -I/usr/include/hdf
  --libs      -> 

  --has-c++   -> no
  --cxx       -> 
  --has-c++4  -> no
  --cxx4      -> 

  --fc        -> gfortran
  --fflags    -> -I/usr/include
  --flibs     -> -lnetcdff  
  --has-f90   -> yes

  --has-dap   -> yes
  --has-nc2   -> yes
  --has-nc4   -> yes
  --has-hdf5  -> yes
  --has-hdf4  -> yes
  --has-pnetcdf-> no

  --prefix    -> /usr
  --includedir-> /usr/include
  --version   -> netCDF 4.3.2
```

**Comment by jswhit** [comment]
Can't reproduce here.

    >>> nc = Dataset('MERRA300.prod.assim.inst3_3d_asm_Cp.20140101.hdf')
    >>> v = nc.variables['SLP']
    >>> v.chunking()
    [1, 144, 288]

[jeff-whitakers-imac-7:~] jsw% python checkversion.py
     netcdf4-python version: 1.1.9
     HDF5 lib version:       1.8.15-patch1
     netcdf lib version:     4.4.0-development
>>>

**Comment by rabernat** [comment]
We fixed this by upgrading to netCDF 4.3.3.1, HDF4 4.2.10, and HDF5 1.8.12.
